### ISSUE TITLE
Attempting to create an Image throws an error 

### ISSUE THREAD

**Issue by mysamdog** [godotengine/godot#16219]
**Godot version:**
Fresh download of Godot 3.0, Mono. Also tested on 3.0 non-Mono, both x64.

**OS/device including version:**
Windows 10 x64, GTX 570, i7-2600k

**Issue description:**
Attempting to create an Image (via "New Image") does nothing other than print
`Set image`
`drivers\gles3\rasterizer_storage_gles3.cpp:836 - Condition ' texture->data_size == 0 && !texture->render_target ' is true. returned: Ref<Image>()`
`drivers\gles3\rasterizer_storage_gles3.cpp:836 - Condition ' texture->data_size == 0 && !texture->render_target ' is true. returned: Ref<Image>()`
`drivers\gles3\rasterizer_storage_gles3.cpp:836 - Condition ' texture->data_size == 0 && !texture->render_target ' is true. returned: Ref<Image>()`
to Output.

Expected outcome
Clicking on "New Image" would create an Image.

**Steps to reproduce:**
Create a new project.
Import a .png (or use the default icon.png)
Create a root Node, and then a Sprite3D
In the Inspector for the Sprite3D, click the dropdown for Texture, and choose "New ImageTexture"
Right click the ImageTexture, and choose edit
Click on the dropdown for Image, and choose "New Image"
An Image is not created, and you get the console output mentioned above.

**Comment by Zylann** [comment]
So here is what happens:

You import a .png, but you never tell Godot to do anything special with it (so you don't need to have a PNG to repro this issue).

Then you tell Godot to create a new `ImageTexture`, which is a resource that needs an `Image` to be used. By default it has no image.

So you create a new `Image`, and... Godot creates a 0x0 image, because that's the default when creating such a resource from nothing. But then the `ImageTexture` gets confused, and doesn't accept that you set an empty image, it doesn't make sense.

Basically, you should have used `Load`, not `New`, because you want to load that PNG, right?
But it turns out your PNG already is imported as a `Texture` by default, so you could have assigned that directly to your `Sprite3D`.
Also, maybe the editor should better indicate what's going on.

**Comment by mysamdog** [comment]
Ah, that makes sense. I initially tried to load the png from within the ImageTexture instead of just using it as a texture directly, which obviously didn't work. I do agree that it could be made more clear in the editor, but it makes sense to me now. Thanks!

**Comment by curioussavage** [comment]
I also had this issue. I could swear that one of the tutorials had me do that or something similar. Seems like ideally the editor should not let you do that all if it never works
